### PR TITLE
add devicetype flag

### LINKED ISSUE #45: Add devicetype env var in cli
**Describe the bug**
Redundant code for emulator and hardware wallet [here](https://github.com/skycoin/hardware-wallet-go/tree/master/cli).

The code for `emulator` and `device` is exactly the same with only one difference i.e., `devicetype`
Which is `deviceWallet.DeviceTypeUsb ` for hardware wallet and `deviceWallet.DeviceTypeEmulator` for emulator

**Possible implementation**
Define `DEVICETYPE` envvar and cli arg for all functions. 
Use this to detect which device type to pass to the functions.

### PR DESCRIPTION
**Pull request by therealssj**
Fixes #45 

Changes:
- Add `-deviceType` flag to set devicetype. Also add `DEVICE_TYPE` env var which can also be used to set the `deviceType`

Does this change need to mentioned in CHANGELOG.md?
yes

### CODE REVIEW

File: cli/address_gen.go [hunk 1]
```diff
@@ -30,33 +31,51 @@ func deviceAddressGenCmd() gcli.Command {
 				Name:  "confirmAddress",
 				Usage: "If requesting one address it will be sent only if user confirms operation by pressing device's button.",
 			},
+			gcli.StringFlag{
+				Name:   "deviceType",
+				Usage:  "Device type to send instructions to, hardware wallet (USB) or emulator.",
+				EnvVar: "DEVICE_TYPE",
+				Value:  "USB",
```
Comment (olemis) [inline cli/address_gen.go]: I do not think implicit default value should be USB . Given the nature of the interactions with the wallet and the fact that sensitive data is transmitted over the wire , I'd rather advice to halt with an error message if value not set or set to invalid value .

**Comment by therealssj** [comment]
I also renamed the cli to `skycoin-hw-cli` and reset version to `0.0.1`

**Comment by olemis** [comment]
> I also renamed the cli to `skycoin-hw-cli` and reset version to `0.0.1`

awesome !

File: CLI.md [hunk 2]
```diff
@@ -36,51 +34,39 @@ $ ./install.sh
...
-   skycoin-cli - the skycoin command line interface
+   skycoin-hw-cli - the skycoin hardware wallet command line interface
 
 USAGE:
-   skycoin-cli [global options] command [command options] [arguments...]
+   skycoin-hw-cli [global options] command [command options] [arguments...]
 
 VERSION:
-   0.24.1
+   0.0.1
```
Comment (olemis) [inline CLI.md]: The previous version n umber (i.e. `0.24.1` was not quite a good decision indeed) . Nonetheless I do not think it should be `0.0.1` . Considering the fact that library operations and semantics are eventually determined by the firmware , in my opinion the version number should be set to match the (minimal) version of the firmware it was built tested against .

File: CLI.md [hunk 3]
```diff
@@ -36,51 +34,39 @@ $ ./install.sh
...
-     emulatorSetPinCode             Configure a PIN code on an emulated device.
-     emulatorWipe                   Ask the emulator to wipe clean all the configuration it contains.
-     emulatorBackup                 Ask the emulator to perform the seed backup procedure.
-     emulatorGetVersion             Ask firmware version.
-     emulatorRecovery               Ask the emulator to perform the seed recovery procedure.
-     emulatorCancel                 Ask the emulator to cancel the ongoing procedure.
-     emulatorTransactionSign        Ask the device to sign a transaction using the provided information.
-     sandbox                        Sandbox.
-     help, h                        Shows a list of commands or help for one command
+     applySettings            Apply settings.
```
Comment (olemis) [inline CLI.md]: @Senyoret1 this has an impact on desktop wallet .

Comment (olemis) [reply CLI.md]: Indeed, maybe not quite ... is the go library used by the desktop wallet ?

File: CLI.md (hunk 2 above)
Comment (olemis) [reply CLI.md]: ... in this case this would be `1.7.0`

File: CLI.md (hunk 3 above)
Comment (therealssj) [reply CLI.md]: nope not used

File: CLI.md (hunk 2 above)
Comment (therealssj) [reply CLI.md]: oh okay TIL.

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Added
 
 - Mnemonic and recovery functions support `--wordCount` argument for the seed size (default `12`) .
+- Add `-deviceType` flag and `DEVICE_TYPE` env var to set devicetype, options are `USB` or `EMULATOR`. Defaults to `USB`.
```
Comment (olemis) [inline CHANGELOG.md]: Note about default value does not hold anymore after 05fff3a .

**Pull request merged**